File: src/main/java/br/com/projetojupiter/model/Criador.java
package br.com.projetojupiter.model;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
@Table(name="tb_criador")
public class Criador {
	
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	private Long id;
	
	@NotNull
	@Size(min=3, max=100)
	private String nome;
	
	@NotNull
	@Size(min=3, max=100)
	@Column(unique=true)
	private String email;
	
	@NotNull
	@Size(min=6, max=100)
	private String senha;
	
	@OneToMany(mappedBy = "criador", cascade = CascadeType.ALL)
	@JsonIgnoreProperties("criador")
	private List<Curso> curso;
	

	public List<Curso> getCurso() {
		return curso;
	}

	public void setCurso(List<Curso> curso) {
		this.curso = curso;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getSenha() {
		return senha;
	}

	public void setSenha(String senha) {
		this.senha = senha;
	}
}

File: src/main/java/br/com/projetojupiter/model/Criador.kt
package br.com.projetojupiter.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import javax.persistence.*
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

@Entity
@Table(name = "tb_criador")
class Criador {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    @NotNull
    @Size(min = 3, max = 100)
    var nome: String = ""

    @Column(unique = true)
    @NotNull
    @Size(min = 3, max = 100)
    var email: String = ""

    @NotNull
    @Size(min = 6, max = 100)
    var senha: String = ""

    @OneToMany(mappedBy = "criador", cascade = [CascadeType.ALL])
    @JsonIgnoreProperties("criador")
    var curso: MutableList<Curso>? = null
}